Recreate Flink recipe after Flink API changes.

dewan-ahmed opened the pull request:
# What changed, and why it matters

This PR recreates the Kafka Flink integration Terraform recipe after the recent Flink API changes. I recreated and tested the Terraform code for this recipe as well as the MermaidJS diagram file.

Please test the recipe and check if it works as expected. I'm looking for input on any improvements on concepts or wording related to the new Flink concepts like application, deployment, etc. I'm also looking for input on possible links to add at the end of the recipe in case the blog from the previous recipe is outdated.

ftisiot commented on docs/tools/terraform/reference/cookbook/kafka-flink-integration-recipe.rst:
```diff
@@ -1,6 +1,226 @@
+      end
+      subgraph Aiven for Apache Flink Application
+      SourceTopic-->|FlinkTableMapping|FlinkSourceTable
+      FlinkSourceTable-->|Filter/Transform|FlinkTargetTable
```
I would put the `Filter/TRansform` as standalone box and have it called `SQL transformation`

ftisiot commented on docs/tools/terraform/reference/cookbook/kafka-flink-integration-recipe.rst:
```diff
@@ -1,6 +1,226 @@
+      end
+
+Imagine that you are collecting CPU usage for hundreds of machines in your data centre and these metrics are populated in an Apache Kafka topic called ``cpu_measurements``. But you're interested in learning about those machines with CPU usages higher than 85% and write the filtered messages into a topic called ``cpu_high_usage``.
+If you relate the above diagram to this example, both source and target Apache Kafka topics are part of the same Apache Kafka cluster. To do the processing on the data, you'll be using an Aiven for Apache Flink Application which is an abstraction layer on top of Apache Flink SQL that includes all the elements related to a Flink job to help build your data processing pipeline. 
```
I would add an extra line in here

ftisiot commented on docs/tools/terraform/reference/cookbook/kafka-flink-integration-recipe.rst:
```diff
@@ -1,6 +1,226 @@
+
+       terraform apply -var-file=var-values.tfvars
+
+The resource ``"aiven_flink"`` creates an Aiven for Apache Flink resource with the project name, choice of cloud, an Aiven service plan, and a specified service name. 
+``"aiven_kafka"`` resource creates an Apache Kafka cluster and two Apache Kafka topics (``cpu_measurements`` and a ``cpu_high_usage``) are created using the ``"aiven_kafka_topic"`` resource.
+Similarly, the ``"aiven_service_integration"`` resource creates the integration between Apache Kafka and the Apache Flink service.
+``aiven_flink_application`` resource ``demo-flink-app`` creates the Aiven for Apache Flink application whereas ``aiven_flink_application_version`` resource ``demo-flink-app-version`` contains all the necessary specifications.
+For example, the application version resource creates two Flink tables, ``iot_measurements_table`` as the source table and ``cpu_high_usage_table`` as the sink table with the specified schema.
```
I would put these as a series of dots, since it's a lot to ingest in few sentences, having it as series of dots could help the reader

ftisiot commented on docs/tools/terraform/reference/cookbook/kafka-flink-integration-recipe.rst:
```diff
@@ -1,6 +1,226 @@
+``aiven_flink_application`` resource ``demo-flink-app`` creates the Aiven for Apache Flink application whereas ``aiven_flink_application_version`` resource ``demo-flink-app-version`` contains all the necessary specifications.
+For example, the application version resource creates two Flink tables, ``iot_measurements_table`` as the source table and ``cpu_high_usage_table`` as the sink table with the specified schema.
+
+Once the Terraform script is run, all of the resources from the manifest are created but you'll need to create the deployment. From the Aiven console, go to the ``Application`` tab under the newly created Aiven for Apache Flink service. Click on ``demo-flink-app`` and click **Create deployment**. If this is your first deployment, you won't have an option for :doc:`savepoint <../../../../products/ ...
```
I would spend a second here to say: "you declared your data pipeline, but the execution is not started. For this you need to create a deployment"

Also regarding savepoint. I would say "If this is your first deployment, you won't have an option to start from a previous state (stored as savepoint)"

ftisiot commented on docs/tools/terraform/reference/cookbook/kafka-flink-integration-recipe.rst:
```diff
@@ -1,6 +1,226 @@
+For example, the application version resource creates two Flink tables, ``iot_measurements_table`` as the source table and ``cpu_high_usage_table`` as the sink table with the specified schema.
+
+Once the Terraform script is run, all of the resources from the manifest are created but you'll need to create the deployment. From the Aiven console, go to the ``Application`` tab under the newly created Aiven for Apache Flink service. Click on ``demo-flink-app`` and click **Create deployment**. If this is your first deployment, you won't have an option for :doc:`savepoint <../../../../products/ ...
+Accept the default setting or make necessary selection and then deploy. 
```
We might also want to say that the deployment can also be done with the Aiven CLI (https://docs.aiven.io/docs/tools/cli/service#avn-service-flink-beta)

dewan-ahmed replied:
Hi @ftisiot I was about to make this change but then realized that the [CLI docs](https://docs.aiven.io/docs/tools/cli/service#avn-service-flink) still refer to the old API concepts like jobs.

> Manages Aiven for Apache Flink® tables and jobs.

> More info on flink table create, flink table delete, flink table get, flink table list, flink job create, flink job cancel, flink job get and flink job list can be found in

I couldn't find https://docs.aiven.io/docs/tools/cli/service#avn-service-flink-beta

ftisiot replied:
Thanks Dewan! I believe you can point people here https://docs.aiven.io/docs/tools/cli/service/flink

ftisiot replied:
Also, I have another PR to fix the list of Flink CLI commands

ftisiot reviewed (changes requested):
Thanks @dewan-ahmed,
Almost perfect, some suggestions in comments

dewan-ahmed commented:
Thanks for the review @ftisiot 🙏🏽 

I addressed all the review feedback and left [one comment](https://github.com/aiven/devportal/pull/1778#discussion_r1115174526). 

Besides, could you please suggest few relevant CTA links for this recipe? I have the usual ones as placeholders but I think we can find more suitable links than the list below:

```
- `Build a Streaming SQL Pipeline with Apache Flink® and Apache Kafka® <https://aiven.io/blog/build-a-streaming-sql-pipeline-with-flink-and-kafka>`_
- `Set up your first Aiven Terraform project <https://docs.aiven.io/docs/tools/terraform/get-started.html>`_
- `Advanced parameters for Aiven for Apache Kafka® <https://docs.aiven.io/docs/products/kafka/reference/advanced-params.html>`_
- `Advanced parameters for Aiven for Apache Flink® <https://docs.aiven.io/docs/products/flink/reference/advanced-params.html>`_

```

ftisiot commented:
Very good work Dewan! I would replace the first link with this https://aiven.io/blog/first-streaming-sql-pipeline-with-apache-flink

ftisiot reviewed (approved):
LGTM, please fix the links and then merge once you're happy!
